Sources/MessagingInApp/Gist/Managers/MessageQueueManager.swift

```diff
@@ -122,17 +122,10 @@ class MessageQueueManager {
 
         let position = message.gistProperties.position
 
-        if let routeRule = message.gistProperties.routeRule {
-            let cleanRouteRule = routeRule.replacingOccurrences(of: "\\", with: "/")
-            if let regex = try? NSRegularExpression(pattern: cleanRouteRule) {
-                let range = NSRange(location: 0, length: Gist.shared.getCurrentRoute().utf16.count)
-                if regex.firstMatch(in: Gist.shared.getCurrentRoute(), options: [], range: range) == nil {
-                    Logger.instance.debug(message: "Current route is \(Gist.shared.getCurrentRoute()), needed \(cleanRouteRule)")
-                    return // exit early to not show the message since page rule doesnt match
-                }
-            } else {
-                Logger.instance.info(message: "Problem processing route rule message regex: \(cleanRouteRule)")
-                return // exit early to not show the message since we cannot parse the page rule for message.
```

Comment:
This PR does not introduce new logic for page rule regex matching. Here is the code that exists in production today. This PR takes this logic, puts it into a re-usable extension in `Message.swift`, and then calls that logic in multiple places.